### PR TITLE
Refactor to new engine-api events api 

### PR DESCRIPTION
**Pull request by jhorwit2**
ref https://github.com/docker/engine-api/pull/315

**\- What I did**

refactored to the new engine-api proposed events api. 

**\- How I did it**

Found all occurrences of the old Events call and replaced it with the new one. 

**\- How to verify it**

Run unit tests and integration tests.

**\- Description for the changelog**

Refactored events api to updated engine-api interface

Signed-off-by: Josh Horwitz horwitzja@gmail.com

cc @vdemeester

### CODE REVIEW

**Comment by jhorwit2** [comment]
[ ] - running / checking out integration tests.

**Comment by vdemeester** [comment]
LGTM 🐸
/cc @stevvooe @cpuguy83 @LK4D4 @runcom

**Comment by vdemeester** [comment]
ping 👼

File: api/client/container/run.go [hunk 1]
```diff
@@ -262,10 +262,15 @@ func runRun(dockerCli *client.DockerCli, flags *pflag.FlagSet, opts *runOptions,
...
-	status := <-statusChan
-	if status != 0 {
-		return cli.StatusError{StatusCode: status}
+	select {
+	case status := <-statusChan:
+		if status != 0 {
+			return cli.StatusError{StatusCode: status}
+		}
+	case err := <-errChan:
+		return fmt.Errorf("Error waiting container's exit code: %v", err)
```
Comment (aaronlehmann) [inline api/client/container/run.go]: "Error waiting for..."?

Also, `error` string should generally start with a lowercase character.

Comment (stevvooe) [inline api/client/container/run.go]: Could probably just use `errors.Wrap` here, as well.

**Comment by stevvooe** [comment]
Design LGTM

**Comment by runcom** [comment]
we are we on this? tests are failing though.

**Comment by jhorwit2** [comment]
I'll take a look at the tests.

File: cli/command/container/run.go [hunk 2]
```diff
@@ -211,10 +211,7 @@ func runRun(dockerCli *command.DockerCli, flags *pflag.FlagSet, opts *runOptions
 		})
 	}
 
-	statusChan, err := waitExitOrRemoved(dockerCli, context.Background(), createResponse.ID, hostConfig.AutoRemove)
-	if err != nil {
-		return fmt.Errorf("Error waiting container's exit code: %v", err)
-	}
+	statusChan, errChan := waitExitOrRemoved(dockerCli, context.Background(), createResponse.ID, hostConfig.AutoRemove)
```
Comment (dnephin) [inline cli/command/container/run.go]: I believe this should use the existing `ctx`, and not create a new one.

Comment (jhorwit2) [inline cli/command/container/run.go]: 👍  i noticed that also while fixing the test. I think this was due to a recent rebase where the context was added. Don't remember one when i first wrote this. 👼

**Comment by vdemeester** [comment]
ping @jhorwit2 👼

**Comment by stevvooe** [comment]
@vdemeester @jhorwit2 Let's try to get this merged. This is a net positive change.

**Comment by jhorwit2** [comment]
@stevvooe @vdemeester sorry for delay. I fixed one of the issues (was with `waitExitOrRemoved`). I should have time tomorrow to figure out the other tests failing.

**Comment by jhorwit2** [comment]
@vdemeester should be all good now! :angel:

**Comment by vdemeester** [comment]
LGTM 🐸
/cc @stevvooe @runcom

**Comment by stevvooe** [comment]
LGTM

**Pull request merged**